test/00_operators/PWelch.cu

```diff
@@ -74,16 +81,45 @@ template <typename TypeParam>
 void helper(PWelchComplexExponentialTest& test)
 {
   MATX_ENTER_HANDLER();
-  test.pb->template InitAndRunTVGenerator<TypeParam>(
-      "00_transforms", "pwelch_operators", "pwelch_complex_exponential", {test.params.signal_size, test.params.nperseg, test.params.noverlap, test.params.nfft, test.params.ftone, test.params.sigma});
+  pybind11::dict cfg(
+    "signal_size"_a=test.params.signal_size,
+    "nperseg"_a=test.params.nperseg,
+    "noverlap"_a=test.params.noverlap,
+    "nfft"_a=test.params.nfft,
+    "ftone"_a=test.params.ftone,
+    "sigma"_a=test.params.sigma,
+    "window_name"_a=test.params.window_name
+  );
+
+  test.pb->template InitAndRunTVGeneratorWithCfg<TypeParam>(
+      "00_operators", "pwelch_operators", "pwelch_complex_exponential", cfg);
 
   tensor_t<TypeParam, 1> x{{test.params.signal_size}};
   test.pb->NumpyToTensorView(x, "x_in");
 
-  // example-begin pwelch-test-1
   auto Pxx  = make_tensor<typename TypeParam::value_type>({test.params.nfft});
-  (Pxx = pwelch(x, test.params.nperseg, test.params.noverlap, test.params.nfft)).run();
-  // example-end pwelch-test-1
+
+  if (test.params.window_name == "none")
+  {
+    (Pxx = pwelch(x, test.params.nperseg, test.params.noverlap, test.params.nfft)).run();
+  }
+  else
+  {
+    auto w = make_tensor<typename TypeParam::value_type>({test.params.nperseg});
+    if (test.params.window_name == "boxcar")
+    {
+      (w = ones<typename TypeParam::value_type>({test.params.nperseg})).run();
```

Comment:
Just FYI `ones` is a dimensionless operator, and you should be able to just write `ones()` and it will adapt to whatever shape it needs to.